### ISSUE TITLE
I have one door but two doors named the same are showing up in HomeKit.

### ISSUE THREAD

**Issue by JohnRucker** [KyleBoyer/homebridge-blaq#2]
**Describe The Bug:**

I have one blaQ hooked up to one of my garage doors.  my ESPHome Version is 2024.7.0. I have the API enabled in the blaQ firmware (selected that when I updated to 2024.7.0) In Home kit I see two garage doors.  Both of the doors can control the blaQ. I can't see how to delete one.  However I can rename them and move them to the correct room. The default name is set to GDO blaQ 613074
**To Reproduce:**

I completely deleted the plugin and let the devices delete out of HomeKit.  I then reinstalled the plugin and configured it and restarted HomeBridge.  Same thing as before two doors appear in HomeKit
**Expected behavior:**

I expect only one door to appear.  
**Logs:**

```
[homebridge-blaq.log.txt](https://github.com/user-attachments/files/16443655/homebridge-blaq.log.txt)
```

**Plugin Config:**

```json
{
    "name": "Homebridge BlaQ",
    "devices": [
        {
            "displayName": "Guest",
            "host": "10.50.0.32",
            "port": 80,
            "mac": "34:b7:da:61:30:74"
        }
    ],
    "garageDoorType": "garage",
    "enableSeparateObstructionSensor": false,
    "enableMotionSensor": false,
    "enablePreCloseWarning": false,
    "enableLockRemotes": true,
    "enableLight": true,
    "enableLearnMode": true,
    "platform": "BlaQHomebridgePlugin"
}
```

**Screenshots:**
![Screenshot 2024-07-31 at 10 08 36 AM](https://github.com/user-attachments/assets/f5ad9a0e-8bd1-42f8-aada-5dde0a005622)

**Environment:**

 **Plugin Version**:  0.2.24
 **Homebridge Version**:  1.8.4 -V 
 **Node.js Version**:  v20.16.0
 **Operating System**:  Raspbian GNU/Linux Bullseye (11)

**Comment by KyleBoyer** [comment]
I'm wondering if the MAC address matching isn't working. Can you check the plugin logs for text containing "Discovered device via" and see what you find?

One thing to try would be, can you uppercase the MAC address in the config? I thought the code accounts for this, but just to double check.

Another thing is, when you uninstalled the plugin, did you choose to remove config and accessory cache? If not, try deleting the accessory cache for entries containing "blaq" in the HomeBridge UI Settings. Then restart the child bridge.
![image](https://github.com/user-attachments/assets/73c7309b-7cca-4b48-a186-88ae97927f7f)

**Comment by JohnRucker** [comment]
While you were typing that I tried one more thing.  I saw in the logs that the auto discovery was finding my blaQ. So I deleted the plugin again and let it remove the doors from homekit.  I then reinstalled the plugin and restarted homebridge.  With no configuration.  I was thinking it would auto discover but it didn't.  So I entered the IP address of my blaQ only (no Mac address).  And everything looks fine now.  I only have one door in HomeKit.  I bet it did have something to do with the MAC address.  I have the blaQ's IP address set with a DHCP reservation so it shouldn't change.  I think I'm good now.  Thanks for the quick response!!

**Comment by KyleBoyer** [comment]
Sounds great! 😁 Feel free to close this issue if you're all set, otherwise I'll do it after a period of inactivity.

**Comment by JohnRucker** [comment]
Thanks again for all the great work!

**Comment by KyleBoyer** [comment]
I do want to mention that the probable fix for the issue you encountered is included in V0.2.25. I did notice that the config MAC was not standardized the same as a discovered device's MAC.

**Comment by JohnRucker** [comment]
Okay great.  I updated to it and put my Mac address back into the config.  Still only see one door (that's good).
